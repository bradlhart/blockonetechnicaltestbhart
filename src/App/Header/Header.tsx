import React from 'react';
import './Header.css';

interface HeaderProps {
  loading: boolean;
  loadBlockInformation: () => void;
}

export const Header: React.FC<HeaderProps> = ({ loading, loadBlockInformation }: HeaderProps) => {
  const handleOnClick = () => loadBlockInformation();

  return (
    <div className="ui two column grid">
      <div className="sixteen wide mobile twelve wide tablet fourteen wide computer column">
        <h2 className="ui header">Block.one Web Application Developer Technical Test</h2>
      </div>
      <div className="sixteen wide mobile four wide tablet two wide computer column" style={{textAlign: 'right'}}>
        <div className={'ui button ' + (loading ? 'loading disabled' : undefined)} onClick={handleOnClick}>
          LOAD
        </div>
      </div>
    </div>
  );
};
