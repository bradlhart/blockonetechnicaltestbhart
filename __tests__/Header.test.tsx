import React from 'react';
import { shallow, ShallowWrapper } from "enzyme";

import Header from '../src/App/Header/Header';

describe('Header', () => {
    let wrapper: ShallowWrapper<any>;
    const props = {
        loading: false,
        loadBlockInformation: jest.fn()
    };

    beforeEach(() => {
        wrapper = shallow(<Header {...props} />);
    });

    test('renders correctly with default props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('renders correctly with props in loading state', () => {
        wrapper.setProps({...props, loading: true});
        expect(wrapper).toMatchSnapshot();
    });

    test('fires function on load button click', () => {
        const button = wrapper.find('[data-testid="loadButton"]');
        button.simulate('click');
        expect(props.loadBlockInformation).toHaveBeenCalled();
    });
});