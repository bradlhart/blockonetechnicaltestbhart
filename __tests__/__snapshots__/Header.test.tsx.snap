// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with default props 1`] = `
<div
  className="ui two column grid"
>
  <div
    className="sixteen wide mobile twelve wide tablet fourteen wide computer column"
  >
    <h2
      className="ui header"
    >
      Block.one Web Application Developer Technical Test
    </h2>
  </div>
  <div
    className="sixteen wide mobile four wide tablet two wide computer column loadButtonAlign"
  >
    <div
      className="ui button "
      data-testid="loadButton"
      onClick={[Function]}
    >
      LOAD
    </div>
  </div>
</div>
`;

exports[`Header renders correctly with props in loading state 1`] = `
<div
  className="ui two column grid"
>
  <div
    className="sixteen wide mobile twelve wide tablet fourteen wide computer column"
  >
    <h2
      className="ui header"
    >
      Block.one Web Application Developer Technical Test
    </h2>
  </div>
  <div
    className="sixteen wide mobile four wide tablet two wide computer column loadButtonAlign"
  >
    <div
      className="ui button loading disabled"
      data-testid="loadButton"
      onClick={[Function]}
    >
      LOAD
    </div>
  </div>
</div>
`;
